dev_docs/kibana_platform_plugin_intro.mdx

```diff
@@ -0,0 +1,271 @@
+
+We don't have great answers to those questions today. Currently, the best answers we have are:
+
+1. Platform plugins are _usually_ plugins that are managed by the Platform Team, but we are starting to see some exceptions.
+2. `core` code contains the most fundamental and stable services needed for plugin development. Everything else goes in a plugin.
+
+We will continue to focus on adding clarity around these types of services and what developers can expect from each.
+
+</DocCallOut>
+
+<DocAccordion buttonContent="A bit of history">
+<DocCallOut title="A bit of history">
+When the Kibana platform and plugin infrastructure was built, we thought of two types of code: core services, and other plugin services. We planned to keep the most stable and fundamental 
+code needed to build plugins inside core.
+
+In reality, we ended up with many platform-like services living outside of core, with no (short term) intention of moving them. We highly encourage plugin developers to use
+them, so we consider them part of platform services.
+
+When we built our platform system, we also thought we'd end up with only a handful of large plugins outside core. Users could turn certain plugins off, to minimize the code
+ footprint and speed up Kibana.
+
+In reality, our plugin model ended up being used like micro-services. Plugins are the only form of encapsulation we provide developers, and they liked it! However, we ended
+ up with a ton of small plugins, that developers never intended to be uninstallable, nor tested in this manner. We are considering ways to provide developers the ability to build services
+ with the encapsulation
+ they desire, without the need to build a plugin.
+
+Another side affect of having many small plugins is that common code often ends up extracted into another plugin. Use case specific utilities are exported,
+ that are not meant to be used in a general manner. This makes our definition of "platform code" a bit trickier to define. We'd like to say "The platform is made up of 
+ every publically exposed service", but in today's world, that wouldn't be a very accurate picture.
+
+We recognize the need to better clarify the relationship between core functionality, platform-like plugin functionality, and functionality exposed by other plugins.
+ It's something we will be working on!
+</DocCallOut>
+</DocAccordion>
+
+The main difference between core functionality and functionality supplied by plugins, is in how it is accessed. Core is
+passed to plugins as the first parameter to their `start` and `setup` lifecycle functions, while plugin supplied functionality is passed as the
+second parameter.  Plugin dependencies must be declared explicitly inside the `kibana.json` file. Core functionality is always provided. Read the 
+section on [how plugins interact with eachother and core](#how-plugins-interact-with-each-other-and-core) for more information.
+
```

Comment:
I think it may be useful to add some more examples of Core services. There's a lot of info in here about plugins but very little about services.

Reply:
Will do, though eventually it'll be great to have those be links to docs with more info on the specific services.